src/extension/index.ts

```diff
@@ -48,6 +52,65 @@ export async function activate(context: ExtensionContext) {
     const panel = new Panel(context, baser, store);
     disposables.push(commands.registerCommand('sarif.showPanel', () => panel.show()));
 
+    // URI handler
+    window.registerUriHandler({
+        async handleUri(uri: vscode.Uri) {
+            if (uri.path === '/alert') {
+                // Launched by Azure DevOps Advanced Security alert page.
+                const pairs = uri.query.split('&');
+                // Find the url parameter.
+                for (const param in pairs) {
+                    const pair = param.split('=');
+                    if (pair[0] === 'url') {
+                        // Decode the alert API URL and pass it to the load function.
+                        await loadAlertSarif(new URL(decodeURIComponent(pair[1])));
+                        break;
+                    }
+                }
+            }
+        }
+    });
+
+    async function loadAlertSarif(url: URL) {
+        try {
+            const session = await vscode.authentication.getSession('microsoft', ['499b84ac-1321-427f-aa17-267ca6975798/.default'], { createIfNone: true });
+            const accessToken = session?.accessToken;
+
+            const response = await fetch(url, {
+                method: 'GET',
+                headers: {
+                    'Accept': 'application/sarif+json',
+                    'Authorization': `Bearer ${accessToken}`,
+                    'User-Agent': 'MS-SarifVSCode.sarif-viewer',
+                }
+            });
+
+            if (response.ok) {
+                // Save the log to a new temp file.
+                const filePath = `${os.tmpdir}\\${(new Date()).getTime()}.sarif`;
+
+                try {
+                    const jsonObject = await response.json();
+                    await fs.promises.writeFile(filePath, jsonObject.value);
+                } catch (error) {
+                    outputChannel.appendLine(`***Exception in loadAlertSarif\n***${error}\n***File path: ${filePath}\n`);
+                }
+
+                // Load the log into the Viewer.
+                store.logs.push(...await loadLogs([Uri.file(filePath)]));
```

Comment:
Should this go inside the `try` block? If there was an error getting the response or writing to disk, then `loadLogs` will probably not be valid either.

Reply:
Good point.